pbj-core/pbj-grpc-helidon/src/main/java/com/hedera/pbj/grpc/helidon/PbjProtocolSelector.java

```diff
@@ -0,0 +1,160 @@
+
+        final var metricRegistry = Metrics.globalRegistry();
+        this.requestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.requests")
+                .scope("vendor")
+                .description("The number of gRPC requests"));
+        this.failedRequestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.request.failures")
+                .scope("vendor")
+                .description("The number of failed gRPC requests"));
+    }
+
+    /**
+     * Called by Helidon to create the sub-protocol for PBJ gRPC requests. The {@link SubProtocolResult} returned
+     * will be responsible for handling the request.
+     */
+    @Override
+    public SubProtocolResult subProtocol(ConnectionContext ctx,
+                                         HttpPrologue prologue,
+                                         Http2Headers headers,
+                                         Http2StreamWriter streamWriter,
+                                         int streamId,
+                                         Http2Settings serverSettings, // unused
+                                         Http2Settings clientSettings, // unused
+                                         StreamFlowControl flowControl,
+                                         Http2StreamState currentStreamState,
+                                         Router router) {
+        this.requestCounter.increment();
+
+        // As per the specification, only POST requests are supported. I would have thought that the code here should
+        // return a response code of 405 (Method Not Allowed) if the method is not POST, but the code here just returns
+        // NOT_SUPPORTED. I'm not sure if this is technically correct.
+        if (prologue.method() != Method.POST) {
+            this.failedRequestCounter.increment();
+            return NOT_SUPPORTED;
+        }
+
+        // Look up the route based on the path. If that route does not exist, we return a 200 OK response with
+        // a gRPC status of NOT_FOUND.
+        final var routing = router.routing(PbjRouting.class, EMPTY);
+        final var route = routing.findRoute(prologue);
+        if (route == null) {
+            this.failedRequestCounter.increment();
+            return new SubProtocolResult(true,
+                    new RouteNotFoundHandler(streamWriter, streamId, currentStreamState));
+        }
+
+        // This is a valid call!
+        return new SubProtocolResult(true,
+                new PbjProtocolHandler(headers,
+                        streamWriter,
+                        streamId,
+                        flowControl,
+                        currentStreamState,
+                        config,
+                        route,
+                        deadlineDetector));
+    }
+
+    /**
+     * A handler for the case where the path is not found.
+     */
+    private static final class RouteNotFoundHandler implements Http2SubProtocolSelector.SubProtocolHandler {
```

Comment:
Minor: where it makes sense, please consider extracting nested classes to the outer level for code maintainability reasons.

Reply:
I extracted this internal class. I didn't see any other candidates for extraction that aren't wrapped up in magic, though